### ISSUE TITLE
`wavappend()` doesn't update header in certain case

### ISSUE THREAD

**Issue by beyondvoid** [dancasimiro/WAV.jl#60]
Hi, everyone. 

It seems that the function `WAV.wavappend` doesn't update the header information in the following case: 

```julia
import WAV 
x1 = rand(16000)
x2 = rand(16000)
WAV.wavwrite(x1, "test.wav", Fs=16000, nbits=16, compression=WAV.WAVE_FORMAT_PCM)
WAV.wavappend(x2, "test.wav")
```

The code above correctly appends the data; however, when opening the file with, e.g., Audacity, the program reads the first part of the wave file containing `x1` only. I think that the header is not updated correctly. It might have something to do with the compression. 

By the way: I'm using Julia-0.6.3 (Windows & Linux).

**Comment by BeyondVoid** [comment]
By the way: reading out the field `subchunk2size` by using 

```julia
header = read("test.wav", 46)
sum(header[41:44] .* (256.^(0:3)))
```

yields `32000`, which is wrong, since it should be `64000`.  (Note: `subchunk2size` equals `number_samples * number_channels * bits_per_sample / 8`).

**Comment by BeyondVoid** [comment]
I just figured out that the value corresponding to the field `chunksize` is correct; however, the value corresponding to `subchunk2size` is wrong. My current workaround is as follows: 

```julia
# Note: `number_samples` and `path_file` is given. 

# Read header. 
file_stream = open(path_file, "r+")
header = read(file_stream, 46)

# Find characters `data` to determine length (44-bytes or 46-bytes) of header. 
# Hex-string below corresponds to the characters `data`.            
range_matched_sequence = search(header, [0x64, 0x61, 0x74, 0x61])
if !isempty(range_matched_sequence)
    indices_data_size = range_matched_sequence + 4;
    if ==(maximum(indices_data_size), 46)
        warn("File features 46-byte header.")
    end
else
    error("Invalid file header.")
end

# Get value that corresponds to field `blockalign`. 
block_align = sum(header[33:34] .* (256.^(0:1)))
if ~isinteger(block_align)
    error("Corrupted file: non-integer block align.")
end

# Update header. 
header[indices_data_size] = reverse(hex2bytes(hex(number_samples*block_align, 8)))

# Overwrite file's header by updated header. 
seekstart(file_stream)
write(file_stream, header)
close(file_stream)
```

This way I avoid reading and writing the whole file.

**Comment by BeyondVoid** [comment]
And by the way: Do you assume 44-byte and/or 46-byte header files?

**Comment by BeyondVoid** [comment]
Hi, everyone. I've found and fixed the bug related to the aforementioned problem. Please go through my pull-request in order to figure out what I've changed.

**Comment by dancasimiro** [comment]
Thanks @BeyondVoid

It’s been a while since I wrote the header code. I’ll take a look and get back to you.

**Comment by BeyondVoid** [comment]
Hi, Daniel. Thanks for taking a look. If you are interested you can find detailed information on the header at [http://soundfile.sapp.org/doc/WaveFormat/](http://soundfile.sapp.org/doc/WaveFormat/).

**Comment by dancasimiro** [comment]
Fixed by pull request #61.
